### ISSUE TITLE
Is this start code valid?

### ISSUE THREAD

**Issue by wbovela** [mwenge/ancipital#1]
I've been trying to get this code to assemble but it fails to load properly. And I think I know why. Admittedly, there's a lot to know about this stuff so I am will be happy to be corrected on this...

Here's the start of the code:
```
* = $0800

;-----------------------------------------------------------------------------------------------------
; SYS 16384 (LaunchGame)
; This launches the program from address $4000, i.e. LaunchGame.
;-----------------------------------------------------------------------------------------------------
; $9E = SYS
; $31,$36,$33,$38,$34,$00 = 16384 ($4000 in hex)
p0800   
        .BYTE $00,$0B,$08,$00,$00,$9E,$31,$36
        .BYTE $33,$38,$34,$00,$00
;---------------------------------------------------------------------------------
; s080D
;---------------------------------------------------------------------------------
s080D
        LDA #$93
        JSR ROM_CHROUT ;$FFD2 - output character                 
        LDA #$8E
        JSR ROM_CHROUT ;$FFD2 - output character
```
This will create a .prg file which loads at $0800 and fill it with the bytes listed from p0800. However, since it ends at $080c and the code starts immediately after what is essentially End Of Basic at $080d the BASIC LOAD command will initialise the Start Of Variables pointer and overwrite the LDA in memory. Since it's not really critical the program will run (start of code is at $4000 anyway) but it's not correct.

Iridis Alpha DID do it correctly:

```
*=$0801
;-------------------------------------------------------
; SYS 16384 ($4000)
; This launches the program from address $4000, i.e. MainControlLoop.
;-------------------------------------------------------
; $9E = SYS
; $31,$36,$33,$38,$34,$00 = 16384 ($4000 in hex)
.BYTE $0C,$08,$0A,$00,$9E,$31,$36,$33,$38,$34,$00

;-------------------------------------------------------
; LaunchCurrentProgram
;-------------------------------------------------------
*=$0810
```

It starts at $0801 ($0800 has to be $00 for the BASIC program to be able to run) and puts the start of code specifically at $0810 which doesn't interfere with anything the BASIC LOAD will put between the End Of Basic and the start of our own code.

Looking at the original ancipital.prg file the bytes at the start of the file look different from the code here. I have the original tape and it loads using NovaLoad so I can only assume that this handles the incoming bytes differently from the BASIC LOAD command (obviously really). 

Bottom line is that I can assemble the code and make it run, but the start of the code at $080d will be mangled bij the Start Of Variables pointer, which has no real effect on the game because the first command simply clears the screen, which probably has little effect once the game is started from its real starting location at $4000.

**Comment by mwenge** [comment]
Hi there,

My understanding is that this part (containing the command SYS 16384) will immediately jump to position $4000 (i.e. 16384 in hex):
```
;-----------------------------------------------------------------------------------------------------
; SYS 16384 (LaunchGame)
; This launches the program from address $4000, i.e. LaunchGame.
;-----------------------------------------------------------------------------------------------------
; $9E = SYS
; $31,$36,$33,$38,$34,$00 = 16384 ($4000 in hex)
p0800   
        .BYTE $00,$0B,$08,$00,$00,$9E,$31,$36
        .BYTE $33,$38,$34,$00,$00

```

At $4000 we have the `LaunchGame` routine which does what it says on the tin!

I've basically bypassed that piece of code in between as there doesn't seem to be a use for it in the game itself - really, what I've done here is choosed `LaunchGame` as an entry point simply because it works! 

As you say the game itself uses a loader, so I'm pretty sure what I've disassembled here is an in-memory dump from VICE rather than the raw contents of a 'tap' file or similar. (I did this for a lot of games, including Iridis Alpha, though some do come without a loader and the raw tape or disk file can be disassembled directly.)

I may be missing your point though - let me know if I am!

**Comment by wbovela** [comment]
I'm sorry. I'm great at being pointless!

You're right, it does work because the first thing that happens is SYS16384 which launches the game.

What I was trying to say is that if the code that starts at $080d was the actual starting code of the game, it wouldn't work as expected, because the the instructions would be overwritten by the BASIC LOAD command.

It's perfectly safe to ignore my ramblings :)
